scripts/verify-contracts.py

```diff
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+import ape
+import click
+import structlog
+from eth_utils import to_checksum_address
+
+
+log = structlog.get_logger(__name__)
+
+
+def _get_contract_info(artifacts_dir: Path) -> dict[str, dict[str, dict[str, str]]]:
+    contract_info: dict[str, dict[str, dict[str, str]]] = {}
+    for file_path in artifacts_dir.glob("*.deployment.json"):
+        with open(file_path, "r") as fp:
+            data = json.load(fp)
+        chain_id = data["chain_id"]
+        if chain_id not in contract_info:
+            contract_info[chain_id] = {}
+        for contract_name, contract in data["contracts"].items():
+            if file_path.stem == "base.deployment":
+                ecosystem = "ethereum"
+            else:
+                _, ecosystem = file_path.stem.split(".")[0].split("-")
+            contract_info[chain_id][contract_name] = {
+                "address": contract["address"],
+                "ecosystem": ecosystem,
+                "network": file_path.parent.stem,
+            }
+
+    return contract_info
+
+
+def _get_rpc_info(rpc_file: Path) -> dict[str, str]:
+    with open(rpc_file, "r") as fp:
+        data = json.load(fp)
+    return data
+
+
+@click.command()
+@click.argument("artifacts_dir", type=click.Path(file_okay=False, dir_okay=True, path_type=Path))
+@click.argument("rpc_file", type=click.Path(file_okay=True, dir_okay=False, path_type=Path))
+def main(artifacts_dir: Path, rpc_file: Path) -> None:
+    contract_info = _get_contract_info(artifacts_dir)
+    rpc_info = _get_rpc_info(rpc_file)
+    for chain_id, contracts in contract_info.items():
+        rpc_url = rpc_info[chain_id]
+        for name, contract in contracts.items():
+            ape_ecosystem = getattr(ape.networks, contract["ecosystem"])
+            ape_network = getattr(ape_ecosystem, contract["network"])
+            try:
+                with ape_network.use_provider(rpc_url):
+                    etherscan = ape.networks.provider.network.explorer
+                    assert etherscan is not None
+                    getattr(ape.project, name).at(contract["address"])
+                    log.info(
+                        "Verifying contract",
+                        contract=name,
+                        chain_id=chain_id,
+                        address=contract["address"],
+                    )
+                    etherscan.publish_contract(to_checksum_address(contract["address"]))
+            except Exception as err:
+                log.error("Failed", err=err)
```

Comment:
Eh, I'm not sure why we should continue in case any exception happens....

Reply:
there is no marking for already verified ones but i can just manually verify remaining 1-2 contract for example at the end

Reply:
Hm, yes, that's a good point.

Could you please then just 
- move the entire `try..except` block to a separate function (.e.g. `_verify_contract` or something like that) to make the whole loop here more readable
- try to narrow the extent of `try..except` scope, in the sense that we probably don't need it around `ape_network.use_provider`

?

Reply:
use_provider should need to be inside try..except as there will be some (zksync) not using etherscan and assert will quit, instead i just want to continue.